.github/workflows/win_ci.yml

```diff
@@ -0,0 +1,260 @@
+name: WIN
+
+on:
+  push:
+    paths-ignore:
+      - 'CHANGES*'
+      - 'Doc/**'
+      - 'appveyor.yml'
+  pull_request:
+    branches: master
+    paths-ignore:
+      - 'CHANGES*'
+      - 'Doc/**'
+      - 'appveyor.yml'
+
+permissions:
+  contents: read
+
+jobs:
+  win_ci:
+    # When continue-on-error is true for an individual build,
+    # that build can fail (it'll show red),
+    # but it won't fail the overall tests
+    continue-on-error: ${{ matrix.continue-on-error || false }}
+
+    # https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md
+    runs-on: ${{ matrix.os || 'windows-2022' }}
+
+    # The name of the test follow the tested language
+    name: >
+      ${{ matrix.SWIGLANG }} ${{ matrix.VER }}
+      ${{ matrix.COMPILER || 'msvc' }} ${{ matrix.os }}
+      ${{ matrix.continue-on-error && '(can fail)' }}
+
+    strategy:
+      matrix:
+        include:
+        - SWIGLANG: csharp
+          INSTALL: 'true'
+        - SWIGLANG: csharp
+          os: 'windows-2019'
+          INSTALL: 'true'
+        - SWIGLANG: java
+          VER: 8
+        - SWIGLANG: java
+          VER: 11
+        - SWIGLANG: java
+          VER: 17
+       #  TODO JDK 21 fails on testing
+       #- SWIGLANG: java
+       #  VER: 21
+        - SWIGLANG: python
+          VER: '3.7'
+        - SWIGLANG: python
+          VER: '3.8'
+        - SWIGLANG: python
+        # VER: '3.9'
+        - SWIGLANG: python
+          VER: '3.10'
+        - SWIGLANG: python
+          VER: '3.11'
+        - SWIGLANG: python
+          VER: '3.12'
+        - SWIGLANG: python
+          INSTALL: 'true'
+          COMPILER: gcc
+      # Run all of them, as opposed to aborting when one fails
+      fail-fast: false
+
+    env:
+      CFLAGS: '-O2'
+      CXXFLAGS: '-O2'
+      CCCL_OPTIONS: '--cccl-muffle /W3 /EHsc'
+      PCRE2_CCCL_LD: '-lpcre2-8-static --cccl-link /NODEFAULTLIB:MSVCRT'
+      CHECK_OPTIONS: 'CSHARPOPTIONS=-platform:x64'
+      SWIGLANG: ${{ matrix.SWIGLANG }}
+      COMPILER: ${{ matrix.COMPILER }}
+      INSTALL: ${{ matrix.INSTALL }}
+      VER: ${{ matrix.VER }}
+      OS: ${{ matrix.os }}
+
+# cl.exe:
+#  https://learn.microsoft.com//cpp/build/reference/compiler-options
+#  /EHc  extern "C" defaults to nothrow.
+#  /EHs  Enable C++ exception handling (no SEH exceptions).
+#  /W3   Warning level.
+#  /WX   Treat warnings as errors.
+#  https://learn.microsoft.com//cpp/build/reference/linker-options
+#  cl.exe pass linker:
+#  /VERBOSE:LIB        Outputs progress messages during the link process.
+#  /NODEFAULTLIB:lib   Ignore library 'lib'
+
+    steps:
+    - name: Machine Info
+      shell: cmd
+      run: |
+          systeminfo | findstr /B /C:"OS Name" /B /C:"OS Version"
+
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        show-progress: true
+
+    - name: Install CCache
+      uses: hendrikmuhs/ccache-action@v1
```

Comment:
Where did you find the information on what this does and how to use it?

Reply:
https://github.com/hendrikmuhs/ccache-action

"A Github action to speedup building using ccache/sccache for C/C++ projects.
Works on Linux, macOS, and Windows."

I do admit:
Not sure it is works.
Not sure it works with MSVC.


GitHub provide [marketplace](https://github.com/marketplace) for action extensions.

Reply:
Arg, I referenced the wrong action, I meant this one:
```
      uses: TheMrMilchmann/setup-msvc-dev@v3
```
and found it here now: https://github.com/TheMrMilchmann/setup-msvc-dev. Anyway, I'm not entirely sure why it is needed if the box already has Visual Studio installed.

Reply:
Regarding ccache working, we could zero the ccache stats at the start of the build (ccache -z -s) and then show the stats at end of the build. If it is working you'll see cache hits. I didn't get involved in the setup of ccache for linux Github Actions, so am not sure if it makes any significant difference.

Reply:
> Arg, I referenced the wrong action, I meant this one:
> 
> ```
>       uses: TheMrMilchmann/setup-msvc-dev@v3
> ```
> 
> and found it here now: https://github.com/TheMrMilchmann/setup-msvc-dev. Anyway, I'm not entirely sure why it is needed if the box already has Visual Studio installed.

This action add the patch and environment variables of the already installed VC C++ and C# compilers.
Cmake does it by itself, our Make file doesn't.
MSVC supplies a batch you call and set the environment.
But with this action is much more convenient.

Reply:
> e could zero the ccache stats at the start of the build (ccache -z -s) and then show the stats at end of the build. If it is working you'll see cache hits. I didn't get involved in the setup of ccache for linux Github Actions, so am not sure if it makes any significant difference.

The CCache takes 3 seconds to setup.
I think it is probably works with GCC, not sure about MSVC.
Anyway, I do not think it makes any damage.

Reply:
I meant path and environment variables. path not patch!